<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NewSimulationDialog</class>
 <widget class="QDialog" name="NewSimulationDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>550</width>
    <height>673</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>550</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>new simulation</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>16</number>
   </property>
   <property name="topMargin">
    <number>16</number>
   </property>
   <property name="rightMargin">
    <number>16</number>
   </property>
   <property name="bottomMargin">
    <number>16</number>
   </property>
   <property name="spacing">
    <number>16</number>
   </property>
   <item row="5" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="title">
      <string>computation settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" columnstretch="0,0,0">
      <item row="1" column="0">
       <widget class="QLabel" name="label_10">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>device</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QRadioButton" name="gpuDeviceRadioButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>CUDA</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="3">
       <widget class="QFrame" name="gpuFrame">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>40</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_5">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <item row="0" column="0" rowspan="2">
          <widget class="QLabel" name="label_11">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>universe size</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QLineEdit" name="gpuUniverseSizeYEdit">
           <property name="text">
            <string>720</string>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="gpuUniverseSizeXEdit">
           <property name="text">
            <string>1440</string>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>x</string>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="label_19">
           <property name="text">
            <string>maximum number of particles</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>maximum number of clusters</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="gpuNumBlocksEdit">
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLineEdit" name="gpuMaxClustersEdit">
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QLineEdit" name="gpuDynamicMemorySizeEdit">
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>number of blocks</string>
           </property>
          </widget>
         </item>
         <item row="9" column="0">
          <widget class="QLabel" name="label_20">
           <property name="text">
            <string>dynamic memory size</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QLineEdit" name="gpuMaxTokensEdit">
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QLineEdit" name="gpuMaxCellsEdit">
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QLineEdit" name="gpuMaxParticlesEdit">
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>maximum number of cells</string>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>maximum number of tokens</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>number of threads per block</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="gpuNumThreadsPerBlockEdit">
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="6" column="0" colspan="3">
       <widget class="QFrame" name="cpuFrame">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>50</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>grid size</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="gridSizeXEdit">
           <property name="text">
            <string>12</string>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>x</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QLineEdit" name="gridSizeYEdit">
           <property name="text">
            <string>6</string>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>unit size</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="unitSizeXEdit">
           <property name="text">
            <string>120</string>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>x</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QLineEdit" name="unitSizeYEdit">
           <property name="text">
            <string>120</string>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_6">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>limit active threads</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="maxThreadsEdit">
           <property name="text">
            <string>8</string>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>120</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_7">
           <property name="minimumSize">
            <size>
             <width>180</width>
             <height>20</height>
            </size>
           </property>
           <property name="text">
            <string>resulting universe size</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="universeSizeXLabel">
           <property name="text">
            <string>1440</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLabel" name="label_8">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>x</string>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QLabel" name="universeSizeYLabel">
           <property name="text">
            <string>720</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_9">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>20</height>
            </size>
           </property>
           <property name="text">
            <string>resulting threads</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLabel" name="activeThreadsLabel">
           <property name="text">
            <string>8 (active)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>/</string>
           </property>
          </widget>
         </item>
         <item row="4" column="3">
          <widget class="QLabel" name="totalThreadsLabel">
           <property name="text">
            <string>72 (total)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QRadioButton" name="cpuDeviceRadioButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>CPU</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>60</height>
      </size>
     </property>
     <property name="title">
      <string>initial configuration</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="2">
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>120</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>energy</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="energyEdit">
        <property name="text">
         <string>0</string>
        </property>
        <property name="frame">
         <bool>false</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>60</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>60</height>
      </size>
     </property>
     <property name="title">
      <string>parameters</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QPushButton" name="simulationParametersButton">
        <property name="text">
         <string>simulation parameters</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="symbolTableButton">
        <property name="text">
         <string>symbol table</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>cpuDeviceRadioButton</tabstop>
  <tabstop>gpuDeviceRadioButton</tabstop>
  <tabstop>gpuUniverseSizeXEdit</tabstop>
  <tabstop>gpuUniverseSizeYEdit</tabstop>
  <tabstop>gpuNumThreadsPerBlockEdit</tabstop>
  <tabstop>gpuNumBlocksEdit</tabstop>
  <tabstop>gpuMaxClustersEdit</tabstop>
  <tabstop>gpuMaxCellsEdit</tabstop>
  <tabstop>gpuMaxTokensEdit</tabstop>
  <tabstop>gpuMaxParticlesEdit</tabstop>
  <tabstop>gpuDynamicMemorySizeEdit</tabstop>
  <tabstop>gridSizeXEdit</tabstop>
  <tabstop>gridSizeYEdit</tabstop>
  <tabstop>unitSizeXEdit</tabstop>
  <tabstop>unitSizeYEdit</tabstop>
  <tabstop>maxThreadsEdit</tabstop>
  <tabstop>energyEdit</tabstop>
  <tabstop>simulationParametersButton</tabstop>
  <tabstop>symbolTableButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>NewSimulationDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>348</x>
     <y>556</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cpuDeviceRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>cpuFrame</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>176</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gpuDeviceRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>cpuFrame</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cpuDeviceRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>gpuFrame</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gpuDeviceRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>gpuFrame</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>234</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>105</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
